src/main/java/fi/digitraffic/tis/vaco/VacoConfiguration.java

```diff
@@ -27,14 +28,20 @@
 @PropertySource(value = "classpath:git.properties", ignoreResourceNotFound = true)
 public class VacoConfiguration {
 
+    private final CredentialsService credentialsService;
+
+    public VacoConfiguration(CredentialsService credentialsService) {
+        this.credentialsService = credentialsService;
+    }
+
```

Comment:
Configuration classes should not have constructors, this probably got autogenerated by some IDEA refactoring. Instead inject the required `CredentialsService` instance as method parameter where it's needed.